expense-splitter/src/components/expense/ExpenseDetail.jsx

```diff
@@ -1,27 +1,135 @@
+import { useEffect, useRef, useState } from "react";
 import { useParams, useNavigate } from "react-router-dom";
 import { UseDataContext } from "../context/SiteContext";
-import Button from "../ui/Button";
+import Card from "../ui/CardExpenseDetail";
 
 function ExpenseDetail() {
-  const { expenses } = UseDataContext();
+  const { expenses, groupData, friends } = UseDataContext();
   const { expenseId } = useParams();
   const navigate = useNavigate();
 
-  const singleExpense = expenses.find((expense) => expense.id === expenseId);
+  // get expense details
+  const expenseDetails = expenses.find((expense) => expense.id === expenseId);
+
+  // get total amount of expense
+  const totalAmount = expenseDetails.amount;
+
+  // get group connected to expense
+  const expenseGroup = groupData.filter(
+    (group) => group.id === expenseDetails.groupId,
+  );
```

Comment:
By simply referencing the element in the array, you can eliminate the steps taken in lines 23-25
ex:

  const expenseGroup = groupData.filter(
    (group) => group.id === expenseDetails.groupId,
  )[0];

Reply:
so I add [0] ?

Reply:
Yes, the [0] will select the first element in the array. Since there will always only be 1 element in the array, it will give you exactly what you want.

If you still wanted to separate it out into another step for readability, you could rewrite line 22 as:
const findGroup = expenseGroup[0]
No need for the find method

Reply:
I see, it deconstructs the array into single object

Reply:
I mean, gets object at index = 0